doc/providers/form.rst

```diff
@@ -132,15 +132,21 @@ form by adding constraints on the fields::
         'translator.domains' => array(),
     ));
 
-    $form = $app['form.factory']->createBuilder('form')
-        ->add('name', 'text', array(
+    // some default data for when the form is displayed the first time
+    $data = array(
+        'name' => 'Your name',
+        'email' => 'Your email',
+    );
+        
+    $form = $app['form.factory']->createBuilder(FormType::class, $data)
+        ->add('name', TextType::class, array(
             'constraints' => array(new Assert\NotBlank(), new Assert\Length(array('min' => 5)))
         ))
-        ->add('email', 'text', array(
+        ->add('email', EmailType::class, array(
             'constraints' => new Assert\Email()
         ))
-        ->add('billing_plan', 'choice', array(
-            'choices' => array(1 => 'free', 2 => 'small_business', 3 => 'corporate'),
+        ->add('billing_plan', ChoiceType::class, array(
+            'choices' => array('free' => 1, 'small business' => 2, 'corporate' => 3),
```

Comment:
All these changes are only for a recent version of Symfony. So, we need to keep both and indicate when to use which.